modules/widgets/simple-payments.php

```diff
@@ -282,6 +297,11 @@ public function validate_ajax_params( $params ) {
 				$errors->add( 'price', __( 'Everything comes with a price tag these days. Please add a your product price.', 'jetpack' ) );
 			}
 
+			$precision = strcmp( $params['currency'], 'JPY' ) === 0 ? 0 : 2;
+			if ( $this->get_decimal_places( $params['price'] ) > $precision ) {
```

Comment:
Can we perhaps store two fields here?

One is the display "price" formatted and visible to the customer and the numerical price that is submitted to us. If we know what currency the customer is using, we can format the display accordingly, so the user can't get it wrong. (For example we'd drop any additional precision in the number or internal string representation).

- How does PayPal expect price format for currencies which may have different groupings/decimal marks?
`1.000,49`
- What currencies does PayPal Support? We'll want to link to that or maybe even document it like we do in https://github.com/Automattic/wp-calypso/blob/master/client/lib/format-currency/currencies.js
- Do we have access to any server currency formatters in Jetpack? We do this sort of thing in our plans endpoints for example.

Reply:
I'd rather not innovate on how we store the data. I agree that we could be doing a better job when showing the price data (not only by adjusting precision, but by using the blog's locale to format the string), but I don't think this is the PR to do it. I'll add a issue for that.

> How does PayPal expect price format for currencies

PayPal expects the price formatted as `#.##`. More specifically, for the [API docs](https://developer.paypal.com/docs/api/payments/v1/#definition-amount): 

> total _string_ **required**
>
> The total amount charged to the payee by the payer. For refunds, represents the amount that the payee refunds to the original payer. Maximum length is 10 characters, which includes:
> - Seven digits before the decimal point.
> - The decimal point.
> - Two digits after the decimal point.

So we are affectively storing the value as we send it over to PayPal.

---
> What currencies does PayPal Support? 

We already have a comment pointing to https://developer.paypal.com/docs/integration/direct/rest/currency-codes/ on https://github.com/Automattic/jetpack/blob/f227d51c7fb986aac8934f119217582c06f36774/modules/widgets/simple-payments.php#L15-L17

---
> Do we have access to any server currency formatters in Jetpack?

I think there used to be a `store` class that provided some methods, but it's long gone. As far as I could find, there is nothing available already on Jetpack.

---

Some extra information I've found (because this is not the first time we play this song) while looking for answers to these questions:

https://github.com/Automattic/jetpack/issues/9131
Thread to discuss payment button data issues (p3fqKv-6th-p2)
Working with multiple currencies (p8RSuw-61-p2)
https://github.com/Automattic/wp-calypso/issues/23213

Reply:
Thanks for the follow up @rodrigoi !